mplc/partner.py

```diff
@@ -70,3 +70,49 @@ def shuffle_labels(self, proportion_shuffled):
                 new_label = shuffle(label)
 
             self.y_train[i] = new_label
+
+
+class PartnerMpl:
+    def __init__(self, partner_parent, mpl):
+        """
+        :type partner_parent: Partner
+        :type mpl: MultiPartnerLearning
+        """
+        self.mpl = mpl
+        self.id = partner_parent.id
+        self.batch_size = partner_parent.batch_size
+        self.minibatch_count = mpl.minibatch_count
+        self.partner_parent = partner_parent
+        self.model_weights = None
+
+        self.minibatched_x_train = np.nan * np.zeros(self.minibatch_count)
+        self.minibatched_y_train = np.nan * np.zeros(self.minibatch_count)
+
+    @property
+    def data_volume(self):
+        return len(self.partner_parent.y_train)
+
+    @property
+    def last_round_score(self):
+        return self.mpl.history.history[self.id]['val_accuracy'][self.mpl.epoch_index, self.mpl.minibatch_index]
+
+    @property
+    def history(self):
+        return self.mpl.history.history[self.id]
+
+    def split_minibatches(self):
+        """Split the dataset of the partner parent in mini-batches"""
+
+        # Create the indices where to split
+        split_indices = np.arange(1, self.minibatch_count + 1) / self.minibatch_count
+
+        # Shuffle the dataset
+        idx = np.random.permutation(len(self.partner_parent.x_train))
+        x_train, y_train = self.partner_parent.x_train[idx], self.partner_parent.y_train[idx]
+
+        # Split the samples and labels
+        self.minibatched_x_train = np.split(x_train, (split_indices[:-1] * len(x_train)).astype(int))
+        self.minibatched_y_train = np.split(y_train, (split_indices[:-1] * len(y_train)).astype(int))
+
+    def build_model(self):
+        return self.mpl.build_model_from_weights(self.model_weights)
```

Comment:
Just a question I just asked myself: is it OK from a python perspective / best practices, to have a method using `self.model_weights` as principal input, but no method actually updating `self.model_weights` from `None` (as these are in other objects)?

Reply:
the function build_model_from_weights is the mpl's one, it needs the model generator provides by the mpl's dataset. 
And the weights are those of the partner... 
Otherwise, I don't really know if it's a proper way to implement that